src/app/dashboard/dashboard.component.ts

```diff
@@ -276,20 +248,268 @@ export class DashBoardComponent implements OnInit {
-            { x: 'Light sleep', y: 15, time: '3:58 am - 5.20 am' }, { x: 'REM', y: 5, time: '5:20 am - 5:47 am' },
-            { x: 'Deep sleep', y: 18, time: '5:47 am - 7:30 am' }
-        ];
-        this.steppiedata = [
-            { x: 'Steps', y: 18, text: '18%' }, { x: 'Legal', y: 8, text: '8%' }
-        ];
-        this.calpiedata = [{ 'x': 'Protein', 'y': 30, 'r': '100', text: '25%' }, { 'x': 'Fat', 'y': 12.5, 'r': '110', text: '12.5%' }, { 'x': 'Fiber', 'y': 12.5, 'r': '115', text: '12.5%' },
-        { 'x': 'Calcium', 'y': 10, 'r': '125', text: '50%' }, { 'x': 'Carbs', 'y': 20, 'r': '125', text: '50%' }, { 'x': 'Vitamins', 'y': 30, 'r': '125', text: '50%' }];
+        this.initializeSkeleton();
+        this.LoadCardData();
+        this.LoadChartData('calorie');
+    }
+    public initializeSkeleton(): void {
+        let cardSkeletons: HTMLCollectionOf<Element> = document.getElementById('card-row').getElementsByTagName('ejs-skeleton');
+        const cards: NodeList = document.querySelectorAll('#card-row .row .e-card');
+        addClass([cards[0] as HTMLElement, cards[1] as HTMLElement, cards[2] as HTMLElement, cards[3] as HTMLElement], "hidden-element-size");
+        (cardSkeletons[0] as HTMLElement).style.height = (cards[0] as HTMLElement).offsetHeight + 'px';
+        (cardSkeletons[1] as HTMLElement).style.height = (cards[1] as HTMLElement).offsetHeight + 'px';
+        (cardSkeletons[2] as HTMLElement).style.height = (cards[2] as HTMLElement).offsetHeight + 'px';
+        (cardSkeletons[3] as HTMLElement).style.height = (cards[3] as HTMLElement).offsetHeight + 'px';
+        removeClass([cards[0] as HTMLElement, cards[1] as HTMLElement, cards[2] as HTMLElement, cards[3] as HTMLElement], "hidden-element-size");
+    }
+    public GetCardData(): any {
+        return new Promise(resolve => setTimeout(() => {
+            let data: { [key: string]: Object } = {};
+            data['calories-eaten'] = '13,100';
+            data['steps-taken'] = '52,100';
+            data['water-consumed'] = '38.7 ltr';
+            data['sleep-duration'] = '50 hr';
+            resolve(data);
+        }, 2000));
+    }
+    public LoadCardData(): void {
+        this.GetCardData().then((data: any) => {
+            document.getElementById('calories-text').textContent = data['calories-eaten'];
+            document.getElementById('steps-text').textContent = data['steps-taken'];
+            document.getElementById('water-text').textContent = data['water-consumed'];
+            document.getElementById('sleep-text').textContent = data['sleep-duration'];
+            let cardSkeletons: HTMLCollectionOf<Element> = document.getElementById('card-row').getElementsByTagName('ejs-skeleton');
+            const cards: NodeList = document.querySelectorAll('#card-row .row .e-card');
+            (cardSkeletons[0] as HTMLElement).style.display = 'none';
```

Comment:
Use Visible property of skeleton instead

Reply:
Implemented Conditional Rendering so it is not needed